testutil/compose/compose/main.go

```diff
@@ -69,40 +70,55 @@ func newRootCmd() *cobra.Command {
 	return root
 }
 
-// newDockerRunFunc returns a cobra run function that generates docker-compose.yml files and executes it.
-func newDockerRunFunc(topic string, dir *string, up *bool, runFunc func(context.Context, string) error) func(cmd *cobra.Command, _ []string) error {
-	return func(cmd *cobra.Command, _ []string) (err error) {
-		ctx := log.WithTopic(cmd.Context(), topic)
+// runFunc defines a function that generates docker-compose.yml from config and returns the template data.
+type runFunc func(context.Context, string, compose.Config) (compose.TmplData, error)
```

Comment:
introduced this API for `define,lock,run`, so that smoke tests can do further modifications to the template data to generate any custom docker-compose.